### PR TITLE
Rename DistributedActors -> DistributedCluster module

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/apple/swift-distributed-actors/issues/1043

### CODE REVIEW

File: Sources/SwiftBenchmarkTools/ArgParser.swift [hunk 1]
```diff
@@ -1,244 +0,0 @@
-//===----------------------------------------------------------------------===//
```
Comment (yim-lee) [inline Sources/SwiftBenchmarkTools/ArgParser.swift:1]: Is the whole `SwiftBenchmarkTools` module getting deleted but GH got confused and not showing that?

**Review by yim-lee — APPROVED** [review]
👍

File: Sources/SwiftBenchmarkTools/ArgParser.swift (hunk 1 above)
Comment (ktoso) [reply Sources/SwiftBenchmarkTools/ArgParser.swift:1]: Yes, the entire module is removed -- if we want to do benchmarks again I'd like to revisit using the new swiftpm plugin based benchmarking tools :)

**Comment by ktoso** [comment]
> 11:01:49 Generating documentation for 'DistributedCluster'... Converting documentation... /code/Sources/DistributedCluster/Docs.docc/Clustering.md:77:3: warning: Topic reference 'ClusterControl/joined(node:within:)-2cla8' couldn't be resolved. No local documentation matches this reference. /code/Sources/DistributedCluster/Docs.docc/Clustering.md:78:3: warning: Topic reference 'ClusterControl/waitFor(_:_:within:)-spiq' couldn't be resolved. No local documentation matches this reference. Conversion complete! (7.72s) Generated DocC archive at '/code/.build/plugins/Swift-DocC/outputs/DistributedCluster.doccarchive'


Fixing... those are so annoying that they're not properly stable :|

**Pull request merged**